presto-tests/src/test/java/io/prestosql/memory/TestMemorySessionProperties.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.memory;
+
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.Session;
+import io.prestosql.plugin.tpch.TpchPlugin;
+import io.prestosql.tests.AbstractTestQueryFramework;
+import io.prestosql.tests.DistributedQueryRunner;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static io.prestosql.SystemSessionProperties.QUERY_MAX_MEMORY_PER_NODE;
+import static io.prestosql.SystemSessionProperties.QUERY_MAX_TOTAL_MEMORY_PER_NODE;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.fail;
+
+public class TestMemorySessionProperties
+        extends AbstractTestQueryFramework
+{
+    public static final Session TINY_SESSION = testSessionBuilder()
+            .setCatalog("tpch")
+            .setSchema("tiny")
+            .build();
+
+    public static final String sql = "SELECT COUNT(*), clerk FROM orders GROUP BY clerk";
+
+    TestMemorySessionProperties()
+    {
+        super(() -> new DistributedQueryRunner.Builder(TINY_SESSION)
+                .setNodeCount(2)
+                .build());
+    }
+
+    @BeforeClass
+    public void init()
+            throws Exception
+    {
+        super.init();
+        try {
+            getQueryRunner().installPlugin(new TpchPlugin());
+            getQueryRunner().createCatalog("tpch", "tpch", ImmutableMap.of());
+        }
+        catch (Exception e) {
+            getQueryRunner().close();
+            throw e;
+        }
+    }
+
+    @Test(timeOut = 240_000)
+    public void testSessionQueryMemoryPerNodeLimit()
+            throws Exception
+    {
+        assertQuery(sql);
+        Session session = testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema("tiny")
+                .setSystemProperty(QUERY_MAX_MEMORY_PER_NODE, "1kB")
+                .build();
+        try {
+            getQueryRunner().execute(session, sql);
+            fail("Expected query to fail due to low query_max_memory_per_node.");
+        }
+        catch (RuntimeException e) {
+            // expected
+        }
+    }
+
+    @Test(timeOut = 240_000)
+    public void testSessionQueryMaxTotalMemoryPerNodeLimit()
+            throws Exception
+    {
+        assertQuery(sql);
+        Session session = testSessionBuilder()
```

Comment:
You can use `Session.builder(getQueryRunner().getDefaultSession())` instead, so you don't need to set catalog and schema